packages/core/src/config.ts

```diff
@@ -235,13 +236,14 @@ const composeExternalsWarnConfig = (
 };
 
 export const composeAutoExternalConfig = (options: {
-  autoExternal: AutoExternal;
+  format: Format;
+  autoExternal?: AutoExternal;
   pkgJson?: PkgJson;
   userExternals?: NonNullable<RsbuildConfig['output']>['externals'];
 }): RsbuildConfig => {
-  const { autoExternal, pkgJson, userExternals } = options;
+  const { format, autoExternal = true, pkgJson, userExternals } = options;
 
-  if (!autoExternal) {
+  if (autoExternal === false || !isIntermediateOutputFormat(format)) {
```

Comment:
Does this mean `umd` and `mf` could never use `autoExternal` anymore?

Reply:
What about make the default value to `true` for esm and cjs，and `false` for others. Instead of ban the field as of now.

Reply:
Still will not work, when format is`"mf"`, it still will directly return. You could use `userExternals` to determine if the config is specified.

Reply:
Got it man.

Reply:
updated the commit, you can review it again.